### ISSUE TITLE
Bug building from source: conflicting types for CINTEnvVars; have struct <anonymous>

### ISSUE THREAD

**Issue by ccalaza** [pyscf/pyscf#1126]
Hi,

I'm currently trying to build PySCF from source in the following 'CentOS Linux release 8.4.2105' environment:
```
GCC-11.2.0
CMake-3.21.1
Python-3.9.6
numpy-1.21.3
scipy-1.7.1
h5py-3.5.0
qcint-5.0.0
libxc-5.1.6
XCFun-2.1.1
```
And I get the following error: 

```
[  1%] Building C object dft/CMakeFiles/xc_itrf.dir/libxc_itrf.c.o
[  2%] Building C object dft/CMakeFiles/xcfun_itrf.dir/xcfun_itrf.c.o
[  5%] Building C object np_helper/CMakeFiles/np_helper.dir/np_helper.c.o
[  5%] Building C object np_helper/CMakeFiles/np_helper.dir/transpose.c.o
[  6%] Building C object np_helper/CMakeFiles/np_helper.dir/condense.c.o
[  7%] Building C object np_helper/CMakeFiles/np_helper.dir/npdot.c.o
[  9%] Building C object np_helper/CMakeFiles/np_helper.dir/omp_reduce.c.o
[ 10%] Building C object np_helper/CMakeFiles/np_helper.dir/pack_tril.c.o
[ 11%] Linking C shared library /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/libxcfun_itrf.so
[ 13%] Linking C shared library /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/libxc_itrf.so
[ 14%] Linking C shared library /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/libnp_helper.so
[ 14%] Built target xcfun_itrf
[ 14%] Built target xc_itrf
[ 14%] Built target np_helper
[ 15%] Building C object mcscf/CMakeFiles/fci.dir/fci_contract_nosym.c.o
[ 17%] Building C object mcscf/CMakeFiles/fci.dir/fci_rdm.c.o
[ 18%] Building C object gto/CMakeFiles/cgto.dir/fill_int2c.c.o
[ 19%] Building C object mcscf/CMakeFiles/fci.dir/fci_contract.c.o
[ 21%] Building C object gto/CMakeFiles/cgto.dir/fill_r_3c.c.o
[ 22%] Building C object mcscf/CMakeFiles/fci.dir/fci_string.c.o
[ 23%] Building C object mcscf/CMakeFiles/fci.dir/fci_4pdm.c.o
[ 25%] Building C object gto/CMakeFiles/cgto.dir/fill_nr_3c.c.o
[ 26%] Building C object mcscf/CMakeFiles/fci.dir/select_ci.c.o
[ 27%] Building C object gto/CMakeFiles/cgto.dir/fill_int2e.c.o
[ 28%] Building C object gto/CMakeFiles/cgto.dir/fill_r_4c.c.o
[ 30%] Building C object gto/CMakeFiles/cgto.dir/ft_ao.c.o
[ 31%] Building C object gto/CMakeFiles/cgto.dir/ft_ao_deriv.c.o
[ 32%] Linking C shared library /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/libfci.so
In file included from /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/gto/ft_ao.c:48:
/dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/gto/ft_ao.h:86:3: error: conflicting types for CINTEnvVars; have struct <anonymous>
   86 | } CINTEnvVars;
      |   ^~~~~~~~~~~
In file included from /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/gto/ft_ao.c:47:
/p/project/cslqip/easybuild/juwels/software/qcint/5.0.0-gcccoremkl-11.2.0-2021.4.0/include/cint.h:257:3: note: previous declaration of CINTEnvVars with type CINTEnvVars
  257 | } CINTEnvVars;
      |   ^~~~~~~~~~~
[ 32%] Built target fci
make[2]: *** [gto/CMakeFiles/cgto.dir/build.make:146: gto/CMakeFiles/cgto.dir/ft_ao.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/gto/ft_ao_deriv.c:22:
/dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/gto/ft_ao.h:86:3: error: conflicting types for CINTEnvVars; have struct <anonymous>
   86 | } CINTEnvVars;
      |   ^~~~~~~~~~~
In file included from /dev/shm/gonzalezcalaza1/juwels/PySCF/2.0.1/GCC-11.2.0/pyscf-2.0.1/pyscf/lib/gto/ft_ao_deriv.c:21:
/p/project/cslqip/easybuild/juwels/software/qcint/5.0.0-gcccoremkl-11.2.0-2021.4.0/include/cint.h:257:3: note: previous declaration of CINTEnvVars with type CINTEnvVars
  257 | } CINTEnvVars;
      |   ^~~~~~~~~~~
make[2]: *** [gto/CMakeFiles/cgto.dir/build.make:160: gto/CMakeFiles/cgto.dir/ft_ao_deriv.c.o] Error 1
[ 34%] Building C object gto/CMakeFiles/cgto.dir/fill_grids_int2c.c.o
make[1]: *** [CMakeFiles/Makefile2:292: gto/CMakeFiles/cgto.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

```

Related issue: https://github.com/pyscf/pyscf/issues/1111

Any idea how to fix this?

Best regards,
Carlos

**Comment by ccalaza** [comment]
It seems like qcint/libcint 5.0.0 introduced breaking changes. Building with qcint 4.4.6 solved the problem. Maybe add this to the documentation here? https://pyscf.org/install.html#compiling-from-source-code

As far as I'm concerned this issue can be closed.

**Comment by sunqm** [comment]
Fixed in PR #1384
